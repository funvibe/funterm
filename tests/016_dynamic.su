lua.values = [1, 2, 3, 4, 5]
lua.result = <<>>

for i = 0, 4 do
    lua.value = lua.values[i]
    lua.result = <<lua.result/binary, lua.value:8>>
end

match lua.result {
    <<a:8, b:8, c:8, d:8, e:8>> -> {
        lua.print("Dynamic construction:", a, b, c, d, e)
    }
}

# Построение с условиями
lua.flags = [true, false, true]
lua.flag_bits = <<>>
for flag in lua.flags: lua.flag_bits = <<lua.flag_bits/bitstring, (flag ? 1 : 0):1>>

lua.flag_bits = <<lua.flag_bits/bitstring, 0:5>>  # Padding до байта

match lua.flag_bits {
    <<f1:1, f2:1, f3:1, pad:5>> -> {
        lua.print("Flags:", f1, f2, f3)
        lua.print("Padding:", pad)
    }
}