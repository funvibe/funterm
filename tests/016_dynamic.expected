Dynamic construction: 1 2 3 4 5
Flags: 1 0 1
Padding: 0